services/distribution/src/test/java/app/coronawarn/server/services/distribution/objectstore/integration/ObjectStoreFilePreservationIT.java

```diff
@@ -0,0 +1,210 @@
+import app.coronawarn.server.services.distribution.objectstore.S3RetentionPolicy;
+import app.coronawarn.server.services.distribution.objectstore.client.S3Object;
+import app.coronawarn.server.services.distribution.runner.Assembly;
+import app.coronawarn.server.services.distribution.runner.RetentionPolicy;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = Application.class, initializers = ConfigFileApplicationContextInitializer.class)
+@DirtiesContext
+@ActiveProfiles("integration-test")
+@Tag("s3-integration")
+class ObjectStoreFilePreservationIT {
+
+  @Autowired
+  private DiagnosisKeyService diagnosisKeyService;
+  @Autowired
+  private Assembly fileAssembler;
+  @Autowired
+  private ApplicationContext applicationContext;
+  @Autowired
+  private S3RetentionPolicy s3RetentionPolicy;
+  @Autowired
+  private ObjectStoreAccess objectStoreAccess;
+  @Autowired
+  private DistributionServiceConfig distributionServiceConfig;
+
+  @MockBean
+  private OutputDirectoryProvider distributionDirectoryProvider;
+
+  @Rule
+  private TemporaryFolder testOutputFolder = new TemporaryFolder();
+
+
+  @BeforeEach
+  public void setup() throws IOException {
+    testOutputFolder.create();
+    objectStoreAccess.deleteObjectsWithPrefix(distributionServiceConfig.getApi().getVersionPath());
+  }
+
+  /**
+   * The test covers a behaviour that manifests itself when data retention and shifting policies cause the file
+   * distribution logic to generate, in subsequent runs, different content for the same timeframes. Below the
+   * distribution problem is described with a concerete daily scenario.
+   * <p>
+   * The test presumes there are 4 consecutive days with 80 keys submitted daily. Running a distribution in Day 4 would
+   * result in: Day 1   -> submission of 80 keys   -> 1 empty file distributed Day 2   -> submission of 80 keys   -> 1
+   * distributed containing 160 keys Day 3   -> submission of 80 keys   -> 1 empty file distributed Day 4   ->
+   * submission of 80 keys   -> 1 distributed containing 160 keys
+   * <p>
+   * All day & hour files already generated should not be changed/removed from S3 even after retention policies have
+   * been applied and a second distribution is triggered.
+   * <p>
+   * If for example, data in Day 1 gets removed completely, then a second distribution run causes a shifting of keys in
+   * different files compared to the previous run, the result being: Day 2   -> submission of 80 keys   -> 1 empty file
+   * generated (different than what is currently on S3) Day 3   -> submission of 80 keys   -> 1 distributed containing
+   * 160 keys (different than 1 empty file on S3) Day 4   -> submission of 80 keys   -> 1 empty file (different than 1
+   * empty file on S3)
+   */
+  @Test
+  void files_once_published_to_objectstore_should_not_be_overriden_because_of_retention_or_shifting_policies()
+      throws IOException {
+
+    //keep data in the past for this test
```

Comment:
`// A leading whitespace makes this comment consistent with the rest of the code base`